diff --git a/rate.module b/rate.module
index f60f1db..254f965 100644
--- a/rate.module
+++ b/rate.module
@@ -108,6 +108,7 @@ function rate_theme($existing, $type, $theme, $path) {
         'use_ajax' => FALSE,
         'can_vote' => FALSE,
         'has_voted' => FALSE,
+        'votes' => NULL,
         'user_voted' => NULL,
         'entity_id' => NULL,
         'entity_type_id' => NULL,
@@ -121,6 +122,7 @@ function rate_theme($existing, $type, $theme, $path) {
         'use_ajax' => FALSE,
         'can_vote' => FALSE,
         'has_voted' => FALSE,
+        'votes' => NULL,
         'user_voted' => NULL,
         'entity_id' => NULL,
         'entity_type_id' => NULL,
@@ -134,6 +136,7 @@ function rate_theme($existing, $type, $theme, $path) {
         'use_ajax' => FALSE,
         'can_vote' => FALSE,
         'has_voted' => FALSE,
+        'votes' => NULL,
         'user_voted' => NULL,
         'entity_id' => NULL,
         'entity_type_id' => NULL,
@@ -147,6 +150,7 @@ function rate_theme($existing, $type, $theme, $path) {
         'use_ajax' => FALSE,
         'can_vote' => FALSE,
         'has_voted' => FALSE,
+        'votes' => NULL,
         'user_voted' => NULL,
         'entity_id' => NULL,
         'entity_type_id' => NULL,
@@ -160,6 +164,7 @@ function rate_theme($existing, $type, $theme, $path) {
         'use_ajax' => FALSE,
         'can_vote' => FALSE,
         'has_voted' => FALSE,
+        'votes' => NULL,
         'user_voted' => NULL,
         'entity_id' => NULL,
         'entity_type_id' => NULL,
diff --git a/src/RateEntityVoteWidget.php b/src/RateEntityVoteWidget.php
index d6e88d7..3f915c9 100644
--- a/src/RateEntityVoteWidget.php
+++ b/src/RateEntityVoteWidget.php
@@ -97,6 +97,13 @@ class RateEntityVoteWidget {
         $entity_id
       );
       $has_voted = (!empty($vote_ids)) ? TRUE : FALSE;
+      $votes = [];
+      if ($has_voted) {
+      /** @var \Drupal\votingapi\Entity\Vote[] $voted */
+        foreach($vote_storage->loadMultiple($vote_ids) as $vote) {
+          $votes[] = $vote->getValue();
+        }
+      }
       $user_can_vote = $this->accountProxy->hasPermission('cast rate vote on ' . $entity_type_id . ' of ' . $bundle);
 
       // Get the current user voted.
@@ -115,6 +122,7 @@ class RateEntityVoteWidget {
         '#use_ajax' => $use_ajax,
         '#can_vote' => $user_can_vote,
         '#has_voted' => $has_voted,
+        '#votes' => $votes,
         '#user_voted' => $user_voted,
         '#entity_id' => $entity_id,
         '#entity_type_id' => $entity_type_id,
